e2e/testcases/oci_sync_test.go

```diff
@@ -522,33 +521,49 @@ func testDigestUpdate(nt *nomostest.NT, image string) {
 }
 */
 
-func getImageDigest(imageName string, publicImage bool) (string, error) {
-	var auth authn.Authenticator
-	if publicImage {
-		auth = authn.Anonymous
-	} else {
-		a, err := google.NewEnvAuthenticator()
```

Comment:
What is the material difference of using `gcloud` rather than oci with this provided authentication method?

Reply:
gcloud supports user auth, not just app credentials.

Before this PR, you would need to run `gcloud auth application-default login` to generate app creds and persist them on disk. Not only is this another required step, but it's also less secure.

After this PR, you can now just use `gcloud auth login`, which users need to have done anyway in order to run the rest of the e2e tests on GKE. gcloud is already used elsewhere in the nomostest framework.

Reply:
SG. Was it generating app creds in CI?

Reply:
CI uses Workload Identity to inject app creds.